ublox_derive/src/output.rs

```diff
@@ -29,6 +29,41 @@ fn generate_debug_impl(pack_name: &str, ref_name: &Ident, pack_descr: &PackDesc)
     }
 }
 
+fn generate_serialize_impl(
+    _pack_name: &str,
+    ref_name: &Ident,
+    pack_descr: &PackDesc,
+) -> TokenStream {
+    let mut fields = vec![];
+    for field in pack_descr.fields.iter() {
+        let field_name = &field.name;
+        let field_accessor = field.intermediate_field_name();
+        let token = if field.map.flatten {
+            quote! {
+                state.serialize_entry(stringify!(#field_name), &self.#field_accessor())?;
+            }
+        } else {
+            quote! {
+                serde::ser::Serialize::serialize(&self.#field_accessor(), serde::__private::ser::FlatMapSerializer(&mut state))?;
```

Comment:
not sure if `serde::__private::ser::FlatMapSerializer` is ok being __private...
as well as the `field.map.flatten` doesn't toggle so will need to look into that some more

but here was the general idea of flattening structs